cmd/rekor-server/app/serve.go

```diff
@@ -112,7 +112,12 @@ var serveCmd = &cobra.Command{
 			os.Exit(1)
 		}
 
-		rekorServer := server.NewServer(tesseraStorage, readOnly, algorithmRegistry)
+		var rekorServer server.RekorServer
+		if viper.GetBool("serve-read-paths") {
+			rekorServer = server.NewReadServer(tesseraStorage, readOnly, algorithmRegistry)
```

Comment:
Need to fix - This breaks with `--read-only` because `tesseraStorage` is nil but is still needed on the read path.